### PR TITLE
Add more hints for map properties config

### PR DESCRIPTION
**Pull request by moarychan**
Fixes https://github.com/Azure/azure-sdk-for-java/issues/26600

Changelog:
- Add hints for map properties keys
- Add final for map attribute in AAD proeprties

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/properties/AzureEventHubsProperties.java [hunk 1]
```diff
@@ -108,7 +108,7 @@ public Processor buildProcessorProperties() {
         propertyMapper.from(this.processor.getConsumerGroup()).to(properties::setConsumerGroup);
 
         propertyMapper.from(this.processor.trackLastEnqueuedEventProperties).to(properties::setTrackLastEnqueuedEventProperties);
-        propertyMapper.from(this.processor.initialPartitionEventPosition).to(properties::setInitialPartitionEventPosition);
+        BeanUtils.copyProperties(processor.initialPartitionEventPosition, properties.getInitialPartitionEventPosition());
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/properties/AzureEventHubsProperties.java:111]: please don't use this, we should stick to the propertyMapper. propertyMapper.from(this.processor.initialPartitionEventPosition).when().to(xxx -> xxx)

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/properties/AzureEventHubsProperties.java [hunk 2]
```diff
@@ -194,7 +194,7 @@ public void setPrefetchCount(Integer prefetchCount) {
          * Map event position to use for each partition if a checkpoint for the partition does not exist in
          * CheckpointStore.
          */
-        private Map<String, StartPosition> initialPartitionEventPosition = new HashMap<>();
+        private final Map<String, StartPosition> initialPartitionEventPosition = new HashMap<>();
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/properties/AzureEventHubsProperties.java:197]: why do we need to change this to final here

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/properties/AzureEventHubsProperties.java (hunk 1 above)
Comment (saragluna) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/properties/AzureEventHubsProperties.java:111]: can we also add some hints for this?

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/properties/AzureEventHubsProperties.java (hunk 1 above)
Comment (moarychan) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/properties/AzureEventHubsProperties.java:111]: I can not find a good way to add a hit for this map property, the same with `authorizationClients` in AAD properties.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/properties/AzureEventHubsProperties.java (hunk 2 above)
Comment (moarychan) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/implementation/eventhubs/properties/AzureEventHubsProperties.java:197]: Due to the map having been initialized, we can make this property final as well. Shall we do this change?

**Review by saragluna — APPROVED** [review]
LGTM.

**Pull request merged**